core/src/martian/encoding.cljc

```diff
@@ -1,25 +1,92 @@
 (ns martian.encoding
-  (:require [clojure.string :as string]))
+  (:require [clojure.string :as str]))
 
-(defn choose-content-type [encoders options]
+;; Media Types
+
+(defn choose-media-type [encoders options]
   (some (set options) (keys encoders)))
 
+(defn get-type-subtype [media-type]
+  (when (and (string? media-type) (not (str/blank? media-type)))
+    (str/trim
+      (if-some [params-sep-idx (str/index-of media-type \;)]
+        (subs media-type 0 params-sep-idx)
+        media-type))))
+
+;; Encoder Selection
+
 (def auto-encoder
   {:encode identity
-   :decode identity
-   :as :auto})
+   :decode identity})
+
+(defn find-encoder [encoders media-type]
+  (or (when-some [type-subtype (get-type-subtype media-type)]
+        (get encoders type-subtype auto-encoder))
+      auto-encoder))
```

Comment:
The old stuff is simplified and properly renamed. (As far as we use `martian.encoding` ns fns for both "Accept" and "Content-Type" encoders, they are about working with "media types" not "content types" per se.)

Reply:
The `auto-encoder` now goes without `:as :auto`, which makes it not so much an "auto" but rather a "missing" encoder.